src/service_cmd/runner/runner.go

```diff
@@ -85,6 +88,9 @@ func (runner *Runner) Run() {
 	}
 
 	srv := server.NewServer("ratelimit", runner.statsStore, localCache, settings.GrpcUnaryInterceptor(nil))
+	runner.mu.Lock()
```

Comment:
Why is this lock necessary? It's not immediately clear to me.

Reply:
Run() and Stop() are called from different threads, since Run() basically hangs forever when it's called in the test.

Since `srv` isn't initialized until Run() is called, seemed like a lock was needed here.

I think I added this lock just because it seemed to make sense for transferring a field between two goroutines. The other lock in server_impl was added because the race detector was failing the test.

Reply:
OK sounds good, thanks.